### PR TITLE
Address #1325 with a CREDITS-LEVELS file.

### LINKED ISSUE #1325: Create a file / sheet that information about each of the 100 levels
Recent conversations in the #help channel and other places have suggested that it would be nice if we had information about each of the 100 levels in a single place.

Currently it's surprisingly difficult to learn about the history of a given level via `git log`. Levels are swapped (as in `git mv`), and they are also renamed (as in `HUSTR` DeHacked strings), so there is no easy identifier to follow their history.

What if we had a sheet (Google Doc?) or text file that had 100 rows with a few helpful columns, like `Name`, `Author`, `Created Date`, `Status`. etc.?

I can take an official automated pass at in, an empty shell that just has the 100 files and empty cells for everything else (or maybe I extract the names from DeHacked).

### PR DESCRIPTION
**Pull request by unithecutie**
(hopefully) Closes #1325.

@selliott512

### CODE REVIEW

**Comment by selliott512** [comment]
Nice. I do have some minor comments about things that do need to be fixed, but I like the format, and you did it quickly. That being said I'm not checking the actual content since, for the most part, I don't know who wrote what (but I did run it through a spell checker - all good).

Here are the issues, which I can help with by editing your commit - I've partially solved them locally anyway.

1. There is a git line ending glitch with file `CREDITS-LEVELS` I think having to do with that file being uploaded to GitHub directly from a Windows computer, and having CRLF line endings (when uploaded via the `git` command line the file gets converted). If you have the command line setup:
```
dos2unix CREDITS-LEVELS
git commit -am 'dos2unix for CREDITS-LEVELS'
```

2. Make sure to add `CREDITS-LEVELS` everywhere `CREDITS-MUSIC` currently is. 

`CREDITS-LEVELS`:
```
git grep -w CREDITS-LEVELS

CREDITS:D: CREDITS-LEVELS, CREDITS-MUSIC, levels, musics, buildcfg.txt, level names
Makefile:CREDITS-LEVELS.txt: CREDITS-LEVELS
Makefile:TEXTDOCS=COPYING.txt CREDITS.txt CREDITS-MUSIC.txt CREDITS-LEVELS.txt
Makefile:	      ./COPYING.txt ./CREDITS.txt ./CREDITS-MUSIC.txt ./CREDITS-LEVELS.txt \
Makefile:	install -Dm 644 CREDITS CREDITS-MUSIC CREDITS-LEVELS NEWS.html README.html -t "$(target)$(docdir)/freedm"
```

`CREDITS-MUSIC`:
```
git grep -w CREDITS-MUSIC

.gitignore:/CREDITS-MUSIC.txt
CREDITS:[See the CREDITS-MUSIC file for specific track authors and titles.]
CREDITS:D: CREDITS-LEVELS, CREDITS-MUSIC, levels, musics, buildcfg.txt, level names
Makefile:CREDITS-MUSIC.txt: CREDITS-MUSIC
Makefile:TEXTDOCS=COPYING.txt CREDITS.txt CREDITS-MUSIC.txt CREDITS-LEVELS.txt
Makefile:	      ./COPYING.txt ./CREDITS.txt ./CREDITS-MUSIC.txt ./CREDITS-LEVELS.txt \
Makefile:	install -Dm 644 CREDITS CREDITS-MUSIC CREDITS-LEVELS NEWS.html README.html -t "$(target)$(docdir)/freedm"
Makefile:	install -Dm 644 CREDITS CREDITS-MUSIC  NEWS.html README.html \
Makefile:	      "$(target)$(docdir)/freedm/CREDITS-MUSIC"                         \
Makefile:	      "$(target)$(docdir)/freedoom/CREDITS-MUSIC"                     
```

3. Don't remove the top level directories in the `uinstall-*` targets. It seems to cause an error. If you have a build setup, particularly on a Linux system, you can test the `install` target to a separate directory, make sure it installs what you expect, then uninstall it via `uninstall`, and see that it deletes everything. Don't worry about this if you just do exactly what `CREDITS-MUSIC` is currently doing but for `CREDITS-LEVELS`. It will probably work, and I can test it.:
```
prefix=/tmp/fd make install 
find /tmp/fd | grep CREDITS-   # CREDITS-MUSIC and CREDITS-LEVELS should be in sync here.
prefix=/tmp/fd make uninstall  # Directory "/tmp/fd" no longer exists.
```

**Comment by unithecutie** [comment]
>  think having to do with that file being uploaded to GitHub directly from a Windows computer

Ah, yeah.
I've allowed edits so now you should be able to.

**Comment by selliott512** [comment]
@unithecutie I took the liberty of applying my suggestions. and squashing the commit to one. If you want to get it back to the way it was prior to my changes that's commit `055ff45d41c86cecd1b6a44d1baed6087799bceb`, where you had run `dos2unix`.

**Comment by selliott512** [comment]
I did check that in the left most column has all 100 levels with no duplicates - great.

One thing about the order - there are two areas where I may have done it different myself, but we can get other opinions

1. It may not be obvious to readers how the secret levels follow their respective levels. I would think that indentation would be helpful here, like in `CREDITS-MUSIC`.
2. The grouping may not be obvious to the casual user, but I know it aligns with the way we break up the levels as per our Discord channels.

If it was me I'd just have a flat list in strictly numeric / ASCII order, with no special handling for secret levels. I'd break into groups by episode for phase 1, and for DM and phase 2 I'd have groups of 10 based on the leading digit (MAP0?, MAP1?, MAP2?, MAP3?, etc.) 

I know you broke them into groups of eight possibly because that's a nice size visually, but having an obvious relationship between groups and numbers may take precedence.

What do you all think?

**Comment by unithecutie** [comment]
> I know it aligns with the way we break up the levels as per our Discord channels.
Indentation would ruin the aligned bars I've used here. I'll re-order them to just be split by 8 for PH2/FDM and 9 for PH1.

**Comment by mc776** [comment]
Frankly I think the formatting for the music was perfect and can easily work here:

```
Episode 1 - Outpost Outbreak
	E1M1: Outer Prison by Gabriel "YukiHerz" Antonio
	E1M2: Communications Center by acc
	E1M3: Waste Disposal by Josef "Paar" Šustek
		E1M9: Nutrient Recycling by Mortrixs19
	E1M4: Supply Depot by GeekMarine
	E1M5: Armory by Oleg "Wereknight" Vovk
	E1M6: Training Facility by Luiz Henrique "Xindage" Gasparin Jeronimo
	E1M7: Xenobiotic Materials Lab by Will "Kid Airbag" Hackney
	E1M8: Outpost Quarry by Luiz Henrique "Xindage" Gasparin Jeronimo
```

**Comment by unithecutie** [comment]
> Frankly I think the formatting for the music was perfect and can easily work here

Yeah, that'll work better. Should the level names be in quotes?

**Comment by unithecutie** [comment]
Off-topic but can we seriously rename E2? The rework in 0.13 is more than enough for it to be renamed, and the name "Military Labs" is very bland compared to the others...

**Review by mc776 — APPROVED** [review]

**Comment by mc776** [comment]
Everything looks good at first glance to me now... I'll leave it open in case anything else needs to be added or updated.

File: CREDITS-LEVELS [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        E4M9: Operations by Matt "RottKing" Cibulas and Ralph "Ralphis" Vickers
+    E4M3: Central Computing by Matt "RottKing" Cibulas and Ralph "Ralphis" Vickers
+    E4M4: Hydroponic Facility by Matt "RottKing" Cibulas and Ralph "Ralphis" Vickers
+    E4M5: Engineering Station by Matt "RottKing" Cibulas and Ralph "Ralphis" Vickers
+    E4M6: Command Center by Matt "RottKing" Cibulas and Ralph "Ralphis" Vickers
+    E4M7: Waste Treatment by Matt "RottKing" Cibulas and Ralph "Ralphis" Vickers
+    E4M8: Launch Bay by Matt "RottKing" Cibulas and Ralph "Ralphis" Vickers
+
+
+
```
Comment (selliott512) [inline CREDITS-LEVELS:48]: You have three blank lines here, but only two for the start of FreeDM.

**Review by selliott512 — COMMENTED** [review]
Looks good to me.

File: CREDITS-LEVELS (hunk 1 above)
Comment (unithecutie) [reply CREDITS-LEVELS:48]: Fixed.

**Pull request merged**